src/lib/Editor/RemoteResource/prepareRequestBody.js

```diff
@@ -0,0 +1,19 @@
+import convertJSONAnnotationToInline from './convertJSONAnnotationToInline'
+
+export default function prepareRequestBody(editedData, format) {
+  if (format === 'json') {
+    return Promise.resolve(JSON.stringify(editedData))
+  } else if (format === 'inline') {
+    return convertJSONAnnotationToInline(editedData)
+      .then((convertedData) => {
+        if (convertedData) {
+          return convertedData
+        } else {
+          return Promise.reject()
+        }
+      })
+      .catch(() => {
+        return Promise.reject()
+      })
```

Comment:
```suggestion
    return convertJSONAnnotationToInline(editedData)
  .then((convertedData) => {
    if (!convertedData) {
      throw new Error("なんかエラーが起きた！");
    }
    return convertedData;
  })
  .catch((error) => {
    console.error("Error in conversion:", error);
  });
```

と書きたかったのかなと思いました。

可能であれば prepareRequestBody をasync関数にしてしまった方が書きやすそうです。

Reply:
async関数にして書き直しました。https://github.com/pubannotation/textae/pull/159/commits/f7e114c9fd663a8d22aaee6a3dbe194e9887d056#diff-d7dd9044379d1ae0ea0a4769065052e329d8db74ba0a262c3cf7a4bcb5c42102